testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2017 Microsoft Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Smoke test to see if Spinnaker can interoperate with Microsoft Azure.
+
+Sample Usage:
+    Assuming you have created $CITEST_ROOT points to the root directory of this repository
+    (which is . if you execute this from the root)
+
+  PYTHONPATH=$CITEST_ROOT:$CITEST_ROOT/spinnaker \
+    python $CITEST_ROOT/spinnaker/azure_smoke_test.py \
+    "--native_hostname=localhost", \
+    "--host_platform=native",
+    "--azure_subscription_id=$AZURE_SUBSCRIPTION_ID", \
+    "--azure_tenant_id=$AZURE_TENANT_ID", \
+    "--azure_client_id=$AZURE_CLIENT_ID", \
+    "--azure_app_key=$AZURE_APP_KEY", \
+    "--test_azure_rg_location=$TEST_AZURE_RG_LOCATION"
+
+"""
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.azure_testing as az
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+import citest.base
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+class AzureSmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the smoke test.
+
+  This scenario defines the different test operations.
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Security Group
+    Delete each of the above (in reverse order)
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 180
+    return agent
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(AzureSmokeTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # create default sec grp name  
+    # Spinnaker-azure appends region to actual RG used by an app to support multi-region
+    # pylint: disable=invalid-name
+    self.TEST_SECURITY_GROUP = 'sec_grp_' + self.bindings['TEST_APP']
+    self.TEST_SECURITY_GROUP_RULE_1=self.TEST_SECURITY_GROUP + '_RULE_1'
+    self.TEST_SECURITY_GROUP_RG = self.bindings['TEST_APP'] + '-' + self.bindings['TEST_AZURE_RG_LOCATION']
```

Comment:
I assume the  GROUP_RULE_1 is an arbitrary name you are picking for this test
and he GROUP_RG is a specific name generated by your clouddriver heuristics.

If this is the case, I suggest you add a method to spinnaker_testing.azure_support that knows this policy and can return the names. it will serve as some documentation, add clarity to future tests, and help distinguish the arbitrary names from the intentional ones.

If you are following standard spinnaker frigga naming and I'm missing that mapping then you can use spinnaker_testing.frigga.Naming to generate the names.